src/sync/app.cpp

```diff
@@ -580,32 +592,37 @@ void App::log_in_with_credentials(const AppCredentials& credentials,
     }, handler);
 }
 
-void App::log_out(std::function<void (Optional<AppError>)> completion_block) const {
+void App::log_out(std::function<void (Optional<AppError>)> completion_block) const
+{
     auto user = current_user();
-    if (!user)
+    if (!user) {
         return completion_block(util::none);
+    }
 
-    std::string route = util::format("%1/session", m_auth_route);
+    std::string route = util::format("%1/auth/session", m_base_route);
 
     auto handler = [completion_block, user](const Response& response) {
+        if (user) {
```

Comment:
If we're going to log out the local user even if there's an error, then shouldn't it just happen immediately rather than waiting for the response?

When would user be falsy here?

Reply:
You're right, I've moved `log_out` to be called before we send the request to the server